docs/source/array-slicing.rst

```diff
@@ -1,20 +1,83 @@
 Slicing
 =======
 
-Dask array supports most of the NumPy slicing syntax.  In particular it
```

Comment:
The diff on this section isn't too informative; https://github.com/TomAugspurger/dask/blob/58c247520373a5006cea6cb9075a59b1398bbc26/docs/source/array-slicing.rst has all of the cases previously, with the `Slicing one dask.array with another dask array` expanded to to (same shape, and boolean mask on first dimension). Actually, I see now that this is out of date, since we can handle the first `N` axes. I'll update.